volatility3/framework/symbols/linux/utilities/modules.py

```diff
@@ -204,138 +223,41 @@ def get_module_info_for_module(
-            for modinfo in results:
-                address = context.layers[kernel.layer_name].canonicalize(modinfo.start)
-                known_modules_addresses.add(address)
-
-        modules_memory_boundaries = cls.get_modules_memory_boundaries(
-            context, kernel_module_name
-        )
-
-        hidden_results = []
-
-        address_mask = context.layers[kernel.layer_name].address_mask
-
-        for module in cls.get_hidden_modules(
-            context,
-            kernel_module_name,
-            known_modules_addresses,
-            modules_memory_boundaries,
-        ):
-            modinfo = cls.get_module_info_for_module(address_mask, module)
-            if modinfo:
-                hidden_results.append(modinfo)
-
-        return hidden_results
-
-    @classmethod
-    def _get_list_modules(
-        cls, context: interfaces.context.ContextInterface, kernel_module_name: str
-    ) -> List[ModuleInfo]:
+    def _validate_gatherers(cls, caller_wanted_gatherers) -> List[str]:
         """
-        Gather `module` instances from lsmod
+        Called by `run_modules_scanners` to validate the caller supplied gatherers list
+        An exception is thrown if an empty gatherers list is given or a list containing an invalid source
         """
-        yield from cls.list_modules(context, kernel_module_name)
-
-    @classmethod
-    def _get_sysfs_modules(
-        cls, context: interfaces.context.ContextInterface, kernel_module_name: str
-    ) -> List[ModuleInfo]:
-        """
-        Gather the `module` instances from sysfs
-        """
-        kernel = context.modules[kernel_module_name]
-
-        sysfs_modules: dict = cls.get_kset_modules(context, kernel_module_name)
-
-        for m_offset in sysfs_modules.values():
-            yield kernel.object(object_type="module", offset=m_offset, absolute=True)
-
-    @classmethod
-    def _validated_sources(cls, caller_wanted_sources) -> List[str]:
-        """
-        Called by `run_modules_scanners` to validate the caller supplied sources list
-        An exception is thrown if an empty source list is given or a list containing an invalid source
-        """
-        if not caller_wanted_sources:
-            raise ValueError("`caller_wanted_sources` must have at least one source.")
-
-        if (
-            len(caller_wanted_sources) == 1
-            and caller_wanted_sources[0] == Modules.source_hidden_identifier
-        ):
+        if not caller_wanted_gatherers:
             raise ValueError(
-                f"{Modules.source_hidden_identifier} cannot be the only source or there is nothing to compare against."
+                "`caller_wanted_gatherers` must have at least one gatherer."
             )
 
-        wanted_sources = []
-
-        for source in caller_wanted_sources:
-            if source not in Modules.all_sources_identifier:
+        for gatherer in caller_wanted_gatherers:
+            if gatherer not in ModuleGatherers.all_gatherers_identifier:
```

Comment:
This prevents people from making their own gatherers, doesn't it?  Either that, or it's inviting people to monkey patch the core gatherer list?

Reply:
Refresh, that was my bad at first, its in the latest version to just subclass() check on the interface